controller/todo_test.go

```diff
@@ -0,0 +1,172 @@
+package controller_test
+
+import (
+	"app/controller"
+	"app/mock"
+	"app/model"
+	"app/router"
+	"app/testdata"
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gin-gonic/gin"
+	"github.com/stretchr/testify/suite"
+)
+
+
+type TodoControllerTestSuite struct {
+    suite.Suite
+    mock   *mock.TodoRepositoryMock
+    router *gin.Engine
+}
+
+func (s *TodoControllerTestSuite) SetupTest() {
+	s.mock = new(mock.TodoRepositoryMock)
+	todoController := controller.NewTodoController(
+		s.mock,
+	)
+	s.router = router.NewRouter(todoController)
+}
+
+
+func (s *TodoControllerTestSuite) TestFindTodo() {
+	s.T().Parallel()
+	testCases := map[string]struct {
+		wantStatusCode int
+	}{
+		"正常系データ": {
+			wantStatusCode: http.StatusOK,
+		},
+	}
+	for name, tc := range testCases {
+		name := name
+		tc := tc
+		s.Run(name, func() {
+			s.T().Parallel()
+			s.mock.On("FindTodo", int64(1)).Return(model.FindTodoResponse{Todo:testdata.Todo}, nil)
+			w := httptest.NewRecorder()
+			req, _ := http.NewRequest("GET", "/todos/1", nil)
```

Comment:
一応以下のような感じでerrにエラーがないことのassertを入れておいても良いかもです。
```suggestion
			req, err := http.NewRequest("GET", "/todos/1", nil)
			s.NoError(err)
```